### DISCUSSION TITLE
ScriptReference/Matrix4x4.Perspective

### DISCUSSION THREAD

**Discussion by giscus** [Developer-Notes-Extension/Unity-Notes#505] [Notes]
# ScriptReference/Matrix4x4.Perspective

https://docs.unity3d.com/ScriptReference/Matrix4x4.Perspective.html

**Comment by Henry00IS** [comment]
This projection matrix is based on the OpenGL convention where forwards is Z positive. Unity uses the DirectX convention where forwards is Z negative. If you want identical behavior to Unity's coordinate system with simple matrix transforms, then make sure to negate the embedded z-flip operation:
```csharp
projection.m02 *= -1f;
projection.m12 *= -1f;
projection.m22 *= -1f;
projection.m32 *= -1f;
```
